### PR TITLE
Reduce number of ExecutionStore updates generated by large scatters [BA-6612 trial]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala [hunk 1]
```diff
@@ -123,7 +123,33 @@ case class WorkflowExecutionActorData(workflowDescriptor: EngineWorkflowDescript
   }
 
   def mergeExecutionDiffs(diffs: Traversable[WorkflowExecutionDiff]): WorkflowExecutionActorData = {
-    diffs.foldLeft(this)((newData, diff) => newData.mergeExecutionDiff(diff))
+    preconsolidateDiffs(diffs) match {
+      case Some(preconsolidated) =>
+        this.mergeExecutionDiff(preconsolidated)
+      case None =>
+        diffs.foldLeft(this)((newData, diff) => newData.mergeExecutionDiff(diff))
+    }
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala:131]: dang that's a cool warning

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala [hunk 2]
```diff
@@ -123,7 +123,33 @@ case class WorkflowExecutionActorData(workflowDescriptor: EngineWorkflowDescript
...
+  }
+
+  def preconsolidateDiffs(diffs: Traversable[WorkflowExecutionDiff]): Option[WorkflowExecutionDiff] = {
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala:134]: Scaladoc plz, not actually sure what this does

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala [hunk 3]
```diff
@@ -123,7 +123,33 @@ case class WorkflowExecutionActorData(workflowDescriptor: EngineWorkflowDescript
...
+    val result = diffs.foldLeft(WorkflowExecutionDiff.empty)((acc, diff) => {
+      if (acc.executionStoreChanges.keySet.intersect(diff.executionStoreChanges.keySet).nonEmpty
+        || acc.jobKeyActorMappings.keySet.intersect(diff.jobKeyActorMappings.keySet).nonEmpty
+        || acc.valueStoreAdditions.keySet.intersect(diff.valueStoreAdditions.keySet).nonEmpty
+        || acc.cumulativeOutputsChanges.intersect(diff.cumulativeOutputsChanges).nonEmpty
+        || acc.rootAndSubworkflowIds.intersect(diff.rootAndSubworkflowIds).nonEmpty
+      ) return None
+      else {
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala:142]: hmm seems like much of what was added is not being tested

**Comment by cjllanwarne** [comment]
Didn't actually work in testing. Closing PR.

**Pull request closed without merging**